lms/djangoapps/courseware/date_summary.py

```diff
@@ -213,6 +213,24 @@ def link(self):
             return ecommerce_service.checkout_page_url(course_mode.sku)
         return reverse('verify_student_upgrade_and_verify', args=(self.course.id,))
 
+    @property
+    def is_enabled(self):
+        """
+        Whether or not this summary block should be shown.
+
+        By default, the summary is only shown if it has date and the date is in the
+        future and the user's enrollment is in upsell modes
+        """
+        is_enabled = super(VerifiedUpgradeDeadlineDate, self).is_enabled
+        enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(self.user, self.course.id)
+
+        # Show the summary if user enrollment is in which allow user to upsell
+        allowed_to_upsell = (
+            is_active and enrollment_mode in CourseMode.UPSELL_TO_VERIFIED_MODES
+        )
+
```

Comment:
if `is_active` is `False` or if both `enrollment_mode` and `is_active` are `None` (i.e., [the enrollment doesn't exist](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/models.py#L1351-L1367)), we'll want to return `True` as well